### ISSUE TITLE
Migrate to Rust 2024 edition

### ISSUE THREAD

**Issue by abey79** [rerun-io/rerun#8428]
At some point, we should migrate to edition 2024.

**Comment by Wumpf** [comment]
Yes! But a bit early, as of now it is not even in stable. It's expected to be released with Rust 1.85 which is expected to be released in February 2025, see https://releases.rs/docs/1.85.0/

**Comment by abey79** [comment]
Thus the `blocked` label ;) I'm using this issue to tag place that need e.g. `let-chains` in the code.

**Comment by emilk** [comment]
It's done!
